python.d/postgres.chart.py

```diff
@@ -208,17 +291,52 @@
             ['table_size', 'size', 'absolute', 1, 1024 * 1024]
         ]},
     'wal': {
-        'options': [None, 'Write-Ahead Logging Statistics', 'files/s', 'write ahead log', 'postgres.wal', 'line'],
+        'options': [None, 'Write-Ahead Logs', 'files', 'wal', 'postgres.wal', 'line'],
+        'lines': [
+            ['written_wal', 'written', 'absolute'],
+            ['recycled_wal', 'recycled', 'absolute'],
+            ['total_wal', 'total', 'absolute']
+        ]},
+    'archive_wal': {
+        'options': [None, 'Archive Write-Ahead Logs', 'files/s', 'archive wal', 'postgres.archive_wal', 'line'],
         'lines': [
             ['file_count', 'total', 'incremental'],
             ['ready_count', 'ready', 'incremental'],
             ['done_count', 'done', 'incremental']
         ]},
-    'background_writer': {
-        'options': [None, 'Checkpoints', 'writes/s', 'background writer', 'postgres.background_writer', 'line'],
+    'checkpointer': {
+        'options': [None, 'Checkpoints', 'writes/s', 'checkpointer', 'postgres.checkpointer', 'line'],
+        'lines': [
+            ['checkpoint_scheduled', 'scheduled', 'incremental'],
+            ['checkpoint_requested', 'requested', 'incremental']
+        ]},
+    'bgwriter': {
+        'options': [None, 'Buffers', 'kilobytes/s', 'bgwriter', 'postgres.bgwriter', 'line'],
+        'lines': [
+            ['buffers_checkpoint', 'checkpoint', 'incremental', 1, 1024],
+            ['buffers_clean', 'clean', 'incremental', 1, 1024],
+            ['maxwritten_clean', 'maxwritten', 'incremental'],
+            ['buffers_backend', 'backend', 'incremental', 1, 1024],
+            ['buffers_alloc', 'alloc', 'incremental', 1, 1024],
+            ['buffers_backend_fsync', 'backend_fsync', 'incremental']
+        ]},
+    'autovacuum': {
+        'options': [None, 'Autovacuum workers', 'workers', 'autovacuum', 'postgres.autovacuum', 'line'],
         'lines': [
-            ['writer_scheduled', 'scheduled', 'incremental'],
-            ['writer_requested', 'requested', 'incremental']
+            ['autovacuum_max_worker', 'max workers', 'absolute'],
+            ['analyze', 'analyze', 'absolute'],
+            ['vacuum', 'vacuum', 'absolute'],
+            ['vacuum_analyze', 'vacuum_analyze', 'absolute'],
+            ['vacuum_freeze', 'vacuum_freeze', 'absolute'],
+            ['brin_summarize', 'brin_summarize', 'absolute']
```

Comment:
Are you sure all that stuff should be in one chart?

> autovacuum_max_workers (integer)

`Specifies the maximum number of autovacuum processes (other than the autovacuum launcher) that may be running at any one time. The default is three. This parameter can only be set at server start.`

why do we need to chart it?

Reply:
> Are you sure all that stuff should be in one chart?
Yes, it shows us autovacuum activity. Then, it give information how to tune autovacuum and other settings...

> (autovacuum_max_workers) why do we need to chart it?
You are right, it is not necessary to chart it. Otherwise we should also chart max_connections. I will remove it.

I will also remove "_" in autovacuum labels.